Add info on creating GH issues.

jotomo opened the pull request:
I noticed that a _/CONTRIBUTING.*_ file will make GitHub hint to that file when someone is about to open an issue:

![contribution-hint](https://cloud.githubusercontent.com/assets/1732305/11454406/22b0120e-962b-11e5-830f-cb762ba379b0.png)

Thought this might be a good opportunity to state how Neo4j uses GH issues.

jotomo committed:
Add info on creating GH issues and PRs.

chrisvest commented:
@jotomo Thanks, this looks like a good start.

@nawroth Do you agree?

@martinfurmanski Do you want to take a look at the build failure? Looks like a flaky test in core-edge.

jotomo commented:
This feature isn't all that new actually: https://github.com/blog/1184-contributing-guidelines
The hint is also displayed when opening a PR, not just when opening an issue.

nawroth commented:
@jotomo It's nice to see that it works with AsciiDoc as well!

I think the most important parts of https://github.com/neo4j-contrib/developer-resources/blob/gh-pages/contribute/contribute.adoc should get copied into the "contributing" document, so people actually see it.

How about this: https://gist.github.com/nawroth/ab4fdd4a1f35869e99f6

chrisvest commented:
Looks like it got hit with a random test failure. I've restarted the build, but if the problem persists you can try rebasing.

jotomo commented:
The idea is to pick up GitHub-minded users who are thinking "I'll just create an issue" and to inform them and redirect if necessary. So I think it would help if it would not be (a too) generic "how to contribute" but to be short and on the GH _level of detail_, like "GitHub issues: bugs only, if you need help see this link (or inlined docs below); for PRs see this link (or inlined docs below). For an overview of (possible) contributions see neo4j.com/develop/contribute. If it's too long/too bread people looking for help might just skim it and open an issue either way, since it may seem more like a diversion than getting help. Those who aren't in such a hurry and don't "just" want to open an issue, but want to contribute, are much more likely to follow a link and read the documentation. So I'm not sure duplicating (and then maintaining) docs is beneficial.
Not sure if I'm getting my point across properly, but one thing I'd really like to be mentioned in the document is when users should open GitHub issues and when not to. Issues asking about help often stay open and unanswered for weeks and that's not helping anybody. Those are the people I had in mind as an audience since they're just opening an issue (in the scenario in the screenshot provided).

nawroth commented:
@jotomo Good points. I updated the gist, WDYT? I want people to see the most important information with the least effort possible!

jotomo commented:
@nawroth Great stuff, conveys the information succinctly :+1:

jotomo commented:
@nawroth I've rebased and updated the branch based on your gist. I've added you as a collaborator to my fork in case you want to change something (if that's a convenient option for you).

nawroth commented:
For obvious reasons, I think this looks good. Great input @jotomo
